scripts/globals/besieged.lua

```diff
@@ -216,14 +216,29 @@ xi.besieged.getAstralCandescence = function()
     return 1 -- Hardcoded to 1 for now
 end
 
-xi.besieged.badges = { 780, 783, 784, 794, 795, 825, 826, 827, 894, 900, 909 }
+xi.besieged.badges =
+{
+    xi.ki.PSC_WILDCAT_BADGE,
+    xi.ki.PFC_WILDCAT_BADGE,
+    xi.ki.SP_WILDCAT_BADGE,
+    xi.ki.LC_WILDCAT_BADGE,
+    xi.ki.C_WILDCAT_BADGE,
+    xi.ki.S_WILDCAT_BADGE,
+    xi.ki.SM_WILDCAT_BADGE,
+    xi.ki.CS_WILDCAT_BADGE,
+    xi.ki.SL_WILDCAT_BADGE,
+    xi.ki.FL_WILDCAT_BADGE,
+    xi.ki.CAPTAIN_WILDCAT_BADGE
+}
 
 xi.besieged.getMercenaryRank = function(player)
     local rank = 0
 
-    for _, v in ipairs(xi.besieged.badges) do
+    for k = #xi.besieged.badges, 1, -1 do
+        local v = xi.besieged.badges[k]
```

Comment:
Minor, but since this value is only used once, I'd place it just in the hasKeyItem call

Reply:
Changed